tests/comments/test_async_serializer.py

```diff
@@ -0,0 +1,229 @@
+    assert comment_3['comment'] == ''
+    assert comment_3['object_pk'] == question.pk
+    assert comment_3['is_removed'] is False
+    assert comment_3['is_censored'] is False
+    assert comment_3['is_blocked'] is True
+    assert len(comment_3['comment_categories']) == 0
+    assert comment_3['last_discussed'] is None
+    assert comment_3['is_moderator_marked'] is False
+    # Shouldn't a blocked comment be moderator marked?
+    assert comment_3['content_type'] == question_ct.pk
+
+    # Testing a comment that has been censored by a moderator:
+    comment_4 = comment_data[3]
+    assert comment_4['id'] == 4
+    assert comment_4['user_name'] == 'unknown user'
+    assert comment_4['user_pk'] == -1
+    assert comment_4['user_profile_url'] == ''
+    assert comment_4['is_deleted'] is True
+    assert comment_4['is_moderator'] is False
+    assert len(comment_4['child_comments']) == 0
+    assert comment_4['comment'] == Comment.objects.get(pk=4).comment
+    assert comment_4['comment'] == ''
+    assert comment_4['object_pk'] == question.pk
+    assert comment_4['is_removed'] is False
+    assert comment_4['is_censored'] is True
+    assert comment_4['is_blocked'] is False
+    assert len(comment_4['comment_categories']) == 0
+    assert comment_4['last_discussed'] is None
+    assert comment_4['is_moderator_marked'] is False
+    assert comment_4['content_type'] == question_ct.pk
+
+    # Testing a comment that has been deleted by its creator:
+    comment_5 = comment_data[2]
+    assert comment_5['id'] == 5
+    assert comment_5['user_name'] == 'unknown user'
+    assert comment_5['user_pk'] == -1
+    assert comment_5['user_profile_url'] == ''
+    assert comment_5['is_deleted'] is True
+    assert comment_5['is_moderator'] is False
+    assert len(comment_5['child_comments']) == 0
+    assert comment_5['comment'] == Comment.objects.get(pk=5).comment
+    assert comment_5['comment'] == ''
+    assert comment_5['object_pk'] == question.pk
+    assert comment_5['is_removed'] is True
+    assert comment_5['is_censored'] is False
+    assert comment_5['is_blocked'] is False
+    assert len(comment_5['comment_categories']) == 0
+    assert comment_5['last_discussed'] is None
+    assert comment_5['is_moderator_marked'] is False
+    assert comment_5['content_type'] == question_ct.pk
+
+    # What is the deleted status of comments that have been
+    # blocked, censored, and removed?
+    # A blocked comment should not be deleted in case it must be unblocked:
+    assert comment_3['is_blocked'] is True
+    assert comment_3['is_deleted'] is False
+
+    # A censored comment is actually is_deleted.
+    assert comment_4['is_censored'] is True
+    assert comment_4['is_deleted'] is True
+    # But does this make sense or should the content remain accessible?
+
+    # A removed comment should be is_deleted:
+    assert comment_5['is_removed'] is True
+    assert comment_5['is_deleted'] is True
```

Comment:
But what do we use the is_deleted for? Does it make sense to put it into the serializer like that? I guess it is only in the serializer, or is it also a property of the model?
I mean, it does make sense that it is `True` for is_censored and is_removed and `False` for is_blocked, because for the first two we delete the comment text, but we do not do that for the latter. But do we need this field in the serializer? What do we use it for in the React code? And if you know what we use it for, does it make sense, that is_blocked is not is_deleted?